### PR TITLE
Unconditionally register Python decomps to Meta key in Python Dispatcher

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85750

This makes them available for Python Dispatcher to service them when
symbolic shapes are involved.  This is needed because under certain
conditions, functionalization will directly call the Meta kernel for a
function in order to produce a properly sized output wrapper tensor
for a view operation. This direct call bypasses the normal decomposition
table mechanism.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

**Comment by ezyang** [comment]
the failing tests here pass on the branch, so I need more stuff...

**Comment by eellison** [comment]
Do you mind if i land https://github.com/pytorch/pytorch/pull/85471 first ?

**Comment by ezyang** [comment]
Pleae do. These look orthogonal, and this PR is failing tests anyway

**Review by wconstab — APPROVED** [review]
I think this makes sense.  Python dispatcher would only pick this meta kernel if there was a matching DK for meta, IIUC.

File: torch/_decomp/__init__.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def add_op_to_table(aten_op):
                 if op_overload in registry:
                     raise RuntimeError(f"duplicate registrations for {op_overload}")
                 registry[op_overload] = fn
+                op_overload.py_impl(torch._C.DispatchKey.Meta)(fn)
```
Comment (bdhirsh) [inline torch/_decomp/__init__.py:103]: Hmmm won't this unconditionally register a bunch of our python decomps to the meta key, and stomp over some of our C++ decomps/meta kernels?

Comment (bdhirsh) [reply torch/_decomp/__init__.py:103]: oh nvm is this fine because the PythonDispatcher needs to be explicitly turned on elsewhere, anyway.

**Pull request closed without merging**